System/Win32/File.hsc

```diff
@@ -345,6 +347,21 @@ getBinaryType name =
     c_GetBinaryType c_name p_btype
   peek p_btype
 
+-- | Get a unique temporary filename.
+--
+-- Calls 'GetTempFileNameW'.
+getTempFileName :: String     -- ^ directory for the temporary file (must be at most MAX_PATH - 14 characters long)
+                -> String     -- ^ prefix for the temporary file name
+                -> Maybe UINT -- ^ if 'Nothing', a unique name is generated
+                              --   otherwise a non-zero value is used as the unique part
+                -> IO String
+getTempFileName dir prefix unique = allocaBytes ((#const MAX_PATH) * sizeOf (undefined :: TCHAR)) $ \c_buf -> do
+  uid <- withFilePath dir $ \c_dir ->
```

Comment:
`uid` is unused and breaking `-Werror`

Reply:
Ah, initially I was considering to return it. But not sure that's useful. Should we?